ClientRuntimes/NodeJS/ClientRuntime.Azure/lib/credentials/userTokenCredentials.js

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information. 
+
+var util = require('util');
+var msrest = require('ms-rest');
+var adal = require('adal-node');
+var Constants = msrest.Constants;
+
+var AzureEnvironment = require('../azureEnvironment');
+
+/**
+* Creates a new UserTokenCredentials object.
+*
+* @constructor
+* @param {string} clientId The active directory application client id. 
+* See {@link https://azure.microsoft.com/en-us/documentation/articles/active-directory-devquickstarts-dotnet/ Active Directory Quickstart for .Net} 
+* for an example.
+* @param {string} domain The domain or tenant id containing this application.
+* @param {string} username The user name for the Organization Id account.
+* @param {string} password The password for the Organization Id account.
+* @param {string} clientRedirectUri The Uri where the user will be redirected after authenticating with AD.
+* @param {object} [options] Object representing optional parameters.
+* @param {AzureEnvironment} [options.environment] The azure environment to authenticate with.
+* @param {string} [options.authorizationScheme] The authorization scheme. Default value is 'bearer'.
+* @param {object} [options.tokenCache] The token cache. Default value is null.
+*/
+function UserTokenCredentials(clientId, domain, username, password, clientRedirectUri, options) {
+  if (!Boolean(clientId) || typeof clientId !== 'string') {
```

Comment:
We will have to change. if user use "new String()", our code will throw by mistake.
Suggest just simplify by verifying the value is set. We might catch less errors, but better than throw on valid cases.